lib/geometrics/plug/open_telemetry.ex

```diff
@@ -29,14 +29,16 @@ defmodule Geometrics.Plug.OpenTelemetry do
   end
 
   def traceparent(otel_ctx) do
-    case :otel_propagator_trace_context.inject(
-           otel_ctx,
-           [],
-           &:otel_propagator_text_map.default_carrier_set/3,
-           []
-         ) do
-      [{"traceparent", traceparent}] -> traceparent
-      [] -> ""
+    :otel_propagator_trace_context.inject(
+      otel_ctx,
+      [],
+      &:otel_propagator_text_map.default_carrier_set/3,
+      []
+    )
+    |> Enum.find(fn {key, _val} -> key == "traceparent" end)
+    |> case do
+      {"traceparent", traceparent} -> traceparent
+      _ -> ""
     end
```

Comment:
I don't have time to test this out at the moment, but I think something like this might be slightly preferable, to find and provide the default in a single pass.

```suggestion
    |> Enum.find_value("", fn
      {"traceparent", traceparent} -> traceparent
      _other -> nil
    end)
```

Reply:
Good suggestion. Pushed an update.

Reply:
Awesome, thanks!